test/cpp/pir/distributed/dist_dialect_test.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2024 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <gtest/gtest.h>
+#include <iostream>
+
+#include "paddle/fluid/pir/dialect/distributed/ir/dist_attribute.h"
+#include "paddle/fluid/pir/dialect/distributed/ir/dist_dialect.h"
+
+using namespace paddle::dialect;  // NOLINT
+
+TEST(process_mesh_test, base) {
+  pir::IrContext* ctx = pir::IrContext::Instance();
+  ctx->GetOrRegisterDialect<DistDialect>();
+  std::vector<int64_t> mesh_shape = {2, 2};
+  std::vector<int64_t> process_ids = {0, 1, 2, 3};
+  std::vector<std::string> dim_names = {"x", "y"};
+  phi::distributed::ProcessMesh process_mesh(
+      mesh_shape, process_ids, dim_names);
+
+  // construct a ProcessMeshAttribute.
+  auto mesh_attr =
+      ProcessMeshAttribute::get(ctx, mesh_shape, process_ids, dim_names);
+  auto mesh_attr_1 = ProcessMeshAttribute::get(ctx, process_mesh);
+  EXPECT_EQ(mesh_attr, mesh_attr_1);
+
+  // test member function.
+  EXPECT_EQ(mesh_attr.process_mesh(), process_mesh);
+  EXPECT_EQ(mesh_attr.shape(), mesh_shape);
+  EXPECT_EQ(mesh_attr.process_ids(), process_ids);
+  EXPECT_EQ(mesh_attr.dim_names(), dim_names);
+  EXPECT_EQ(mesh_attr.size(), 4);
+  EXPECT_EQ(mesh_attr.ndim(), 2);
+  EXPECT_EQ(mesh_attr.dim_size(0), 2);
+  EXPECT_EQ(mesh_attr.dim_size("y"), 2);
+  EXPECT_FALSE(mesh_attr.empty());
+  EXPECT_TRUE(mesh_attr.contains(3));
+  EXPECT_EQ(mesh_attr.hash(), process_mesh.hash());
+  EXPECT_EQ(mesh_attr.to_string(), process_mesh.to_string());
+}
+TEST(tensor_dist_attr_test, base) {
+  pir::IrContext* ctx = pir::IrContext::Instance();
+  ctx->GetOrRegisterDialect<DistDialect>();
+
+  // std::vector<int64_t> x_shape = {32, 48};
+  std::vector<int64_t> mesh_shape = {2, 3};
+  std::vector<int64_t> process_ids = {0, 1, 2, 3, 4, 5};
+  std::vector<std::string> dim_names = {"x", "y"};
+  phi::distributed::ProcessMesh process_mesh(
+      mesh_shape, process_ids, dim_names);
+  std::vector<int64_t> dims_mapping = {0, -1};
+  paddle::flat_hash_map<int64_t, phi::ReduceType> partial_status;
```

Comment:
add more test that if partial_status is not equal, the tensor_dist_attr is not equal.

Reply:
done, thanks!